### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by nickfrisch**

### CODE REVIEW

File: src/main/cpp/gamepad/TeleopControl.cpp [hunk 1]
```diff
@@ -142,7 +142,13 @@ TeleopControl::TeleopControl() : m_axisIDs(),
 		m_buttonIDs[SHOOTER_PREPARE_TO_SHOOT] = IDragonGamePad::POV_0;	
 	
 		m_controllerIndex[TURRET_LIMELIGHT_AIM] = ctrlNo;
-		m_buttonIDs[TURRET_LIMELIGHT_AIM] = IDragonGamePad::LEFT_BUMPER;	
+		m_buttonIDs[TURRET_LIMELIGHT_AIM] = IDragonGamePad::LEFT_BUMPER;
+
+		m_controllerIndex[HOLD] = ctrlNo;
+		m_buttonIDs[HOLD] = IDragonGamePad::LEFT_TRIGGER_PRESSED;
+
+		m_controllerIndex[INTAKE] = ctrlNo;
```
Comment (witcpalek) [inline src/main/cpp/gamepad/TeleopControl.cpp]: This isn't a telop function

File: src/main/cpp/gamepad/TeleopControl.cpp [hunk 2]
```diff
@@ -142,7 +142,13 @@ TeleopControl::TeleopControl() : m_axisIDs(),
 		m_buttonIDs[SHOOTER_PREPARE_TO_SHOOT] = IDragonGamePad::POV_0;	
 	
 		m_controllerIndex[TURRET_LIMELIGHT_AIM] = ctrlNo;
-		m_buttonIDs[TURRET_LIMELIGHT_AIM] = IDragonGamePad::LEFT_BUMPER;	
+		m_buttonIDs[TURRET_LIMELIGHT_AIM] = IDragonGamePad::LEFT_BUMPER;
+
+		m_controllerIndex[HOLD] = ctrlNo;
```
Comment (witcpalek) [inline src/main/cpp/gamepad/TeleopControl.cpp]: This is the same as prepare to shoot

File: src/main/cpp/states/ballhopper/BallHopperStateMgr.cpp [hunk 3]
```diff
@@ -136,6 +137,19 @@ BallHopperStateMgr::BallHopperStateMgr() : m_currentState(nullptr),
 /// @return void
 void BallHopperStateMgr::RunCurrentState()
 {
+    //process teleop/manual interrupts
```
Comment (witcpalek) [inline src/main/cpp/states/ballhopper/BallHopperStateMgr.cpp]: Ball hopper states should be driven by the shooter states

**Review by witcpalek — CHANGES_REQUESTED** [review]

File: src/main/cpp/states/ballhopper/BallHopperStateMgr.cpp (hunk 3 above)
Comment (nickfrisch) [reply src/main/cpp/states/ballhopper/BallHopperStateMgr.cpp]: These changes were being used for testing the shooter.  I don't think the ballhopper was tied to the shooter buttons so this was just a bandaid fix while we were there and we will have to make sure the ballhopper is being controlled by the shooter states.

Comment (witcpalek) [reply src/main/cpp/states/ballhopper/BallHopperStateMgr.cpp]: This block should have caused the hopper to go into hold state; wasn't it working?      

  if ( controller->IsButtonPressed( TeleopControl::FUNCTION_IDENTIFIER::SHOOTER_PREPARE_TO_SHOOT ))
        {
            Logger::GetLogger()->ToNtTable(ntName, "Current State", "Prepare to Shoot");
            SetCurrentState( SHOOTER_STATE::GET_READY, false );
            BallHopperStateMgr::GetInstance()->SetCurrentState( BallHopperStateMgr::HOLD, false);
            BallTransferStateMgr::GetInstance()->SetCurrentState( BallTransferStateMgr::BALL_TRANSFER_STATE::TO_SHOOTER, false );
        }

Similarly, the particular color shooting spots right now indicate rapid release and probably should be slow release.

Comment (nickfrisch) [reply src/main/cpp/states/ballhopper/BallHopperStateMgr.cpp]: I wasn't the one controlling and testing the shooter but I don't think they were pressing the prepare to shoot button.  From the testing yesterday, I'm not sure if they want us to use the slow release function.  It sounded like they wanted to release all the balls at once for better accuracy.  I can figure out what's going on tonight and sort out this pull request.

Comment (witcpalek) [reply src/main/cpp/states/ballhopper/BallHopperStateMgr.cpp]: ok, sounds good

**Review by witcpalek — APPROVED** [review]

**Pull request merged**